### PR TITLE
Generate initial CP1250 and CP1251 button fonts

### PR DESCRIPTION
**Pull request by zenseii**
This generates letters for the CP1252 encoding, which includes fonts for Polish, Hungarian and Czech.
Only Ś and Ć are made, the rest are replaced with their closest Latin couterparts since only those two letters are currently required by the translations we have.

Initial CP1251 is made by replacing with closest Latin alphabet counterparts and doing small edits to some of them.

CP1250, Polish:
image

CP1251, Cyrillic:
image

image

### CODE REVIEW

File: src/fheroes2/gui/ui_font.cpp [hunk 1]
```diff
@@ -2506,7 +2506,245 @@ namespace
         fheroes2::SetPixel( released[58], offset + 0, offset + 1, buttonGoodReleasedColor );
         fheroes2::SetPixel( released[58], offset + 8, offset + 8, buttonGoodReleasedColor );
     }
+    void generateGoodCP1250ButtonFont( std::vector<fheroes2::Sprite> & released )
```
Comment (ihhub) [inline src/fheroes2/gui/ui_font.cpp:2509]: Please add an empty line before this function to have a visual separation of functions.

File: src/fheroes2/gui/ui_font.cpp [hunk 2]
```diff
@@ -2506,7 +2506,245 @@ namespace
...
+        // U with diaerisis. Only copied from U.
+        released[188] = released[53];
 
+        // Y with acute. Only copied from Y.
+        released[189] = released[57];
+
+        // T with cedilla. Only copied from T.
+        released[190] = released[52];
+    }
+    void generateGoodCP1251ButtonFont( std::vector<fheroes2::Sprite> & released )
```
Comment (ihhub) [inline src/fheroes2/gui/ui_font.cpp:2650]: Please add an empty line before this function to have a visual separation of functions.

File: src/fheroes2/gui/ui_font.cpp [hunk 3]
```diff
@@ -2506,7 +2506,245 @@ namespace
...
+        released[181] = released[56];
+
+        // Mirrored R, Cyrillic YA
+        released[191] = released[50];
+        {
+            fheroes2::Sprite temp = fheroes2::Flip( released[191], true, false );
+            fheroes2::Copy( temp, 0, 0, released[191], 0, 0, temp.width(), temp.height() );
+        }
+    }
     void generateGoodCP1252ButtonFont( std::vector<fheroes2::Sprite> & released )
```
Comment (ihhub) [inline src/fheroes2/gui/ui_font.cpp:2748]: Please add an empty line before this function to have a visual separation of functions.

File: src/fheroes2/gui/ui_font.cpp [hunk 4]
```diff
@@ -2633,16 +2871,19 @@ namespace fheroes2
         // NOTE: As soon as code structure is agreed on functions for all Code Pages will be added.
         switch ( language ) {
         case SupportedLanguage::Polish:
-            // generateGoodCP1250ButtonFont( icnVsSprite );
+            generateGoodCP1250ButtonFont( icnVsSprite[ICN::BUTTON_GOOD_FONT_RELEASED] );
```
Comment (ihhub) [inline src/fheroes2/gui/ui_font.cpp:2874]: To keep naming the same I propose to use `generateCPXXXGoodButtonFont()` as we did for `generateBaseButtonFont()`.

File: src/fheroes2/gui/ui_font.cpp [hunk 5]
```diff
@@ -2506,7 +2506,245 @@ namespace
...
+        // B
+        released[162] = released[34];
+
+        // E
+        released[165] = released[37];
+
+        // Mirrored N, Cyrillic I
+        released[168] = released[46];
+        {
+            fheroes2::Sprite temp = fheroes2::Flip( released[168], true, false );
```
Comment (ihhub) [inline src/fheroes2/gui/ui_font.cpp:2705]: I will add an overloaded version of `Flip()` function to avoid this way of doing thigs.

**Review by ihhub — CHANGES_REQUESTED** [review]
Hi @zenseii , I put several comments here. Would you mind to take a look at them in your spare time?

**Comment by ihhub** [comment]
Hi @zenseii , is this pull request ready for review?

**Comment by zenseii** [comment]
@ihhub, this PR can be review now or when the checks have finished, and of course when you should feel like doing it.

**Review by ihhub — APPROVED** [review]

**Comment by ihhub** [comment]
@zenseii , a bunch of thanks for improving this part of the code!

**Pull request merged**